src/Lorentz/Contracts/BaseDAO/Management.hs

```diff
@@ -0,0 +1,79 @@
+-- SPDX-FileCopyrightText: 2020 TQ Tezos
+-- SPDX-License-Identifier: LicenseRef-MIT-TQ
+
+module Lorentz.Contracts.BaseDAO.Management
+  ( acceptOwnership
+  , transferOwnership
+  ) where
+
+import Lorentz
+import Lorentz.Contracts.BaseDAO.Types
+import Lorentz.Contracts.ManagedLedger.Doc (DRequireRole(..))
+
+-- | Set the value of pending owner in storage
+-- using the address in parameter.
+transferOwnership
+  :: Entrypoint TransferOwnershipParam Storage
+transferOwnership = do
+  dip authorizeAdmin
+
+  -- Check if new owner is current admin. If so, we just store a `none`
+  -- value to the pending owner field in the storage, and leave a False
+  -- value on stack so that the rest of operation can be skipped. If not
+  -- we just leave a True value to indicate that setting of new pending
+  -- owner should continue.
+  fromNamed #newOwner
+  dup
+  dip $ do
+    dip $ stGetField #sAdmin
+
+    eq
+    if_ (do none; stSetField #sPendingOwner; push False) (push True)
```

Comment:
IMO, it would look better if you used `if IsEq then …` here, but that's up to you.